### ISSUE TITLE
Multi faces tracking bug

### ISSUE THREAD

**Issue by eric-beaufol** [zappar-xr/zappar-threejs#85]
**Describe the bug**
When first detected face leaves the screen, second face is no more tracked.
When first face enters back to screen, second is tracked again, looks like an index collapse problem.

**To Reproduce**
```
let manager = new ZapparThree.LoadingManager();
const faceAnchorGroupsArr = []
let currentSize

let renderer = new THREE.WebGLRenderer();
document.body.appendChild(renderer.domElement);

renderer.setSize(800, 500);

let camera = new ZapparThree.Camera();
camera.userCameraMirrorMode = ZapparThree.CameraMirrorMode.Poses;
camera.poseMode = ZapparThree.CameraPoseMode.Attitude
ZapparThree.glContextSet(renderer.getContext());

let scene = new THREE.Scene();
scene.background = camera.backgroundTexture;


ZapparThree.permissionRequestUI().then(function(granted) {
  if (granted) {
    camera.start(true);
  } else {
    ZapparThree.permissionDeniedUI();
  }
});

let faceTracker = new ZapparThree.FaceTrackerLoader(manager).load();
faceTracker.maxFaces = 2

faceTracker.onNewAnchor.bind(anchor => {
  let faceAnchorGroup = new ZapparThree.FaceAnchorGroup(camera, faceTracker, anchor.id)
  scene.add(faceAnchorGroup)
  faceAnchorGroupsArr.push(faceAnchorGroup)

  let box = new THREE.Mesh(
    new THREE.BoxBufferGeometry(),
    new THREE.MeshBasicMaterial({ color: faceAnchorGroupsArr.length > 1 ? 0xffffff : 0x000000 })
  );
  box.scale.set(0.1, 0.1, 0.1);
  box.position.set(0, 0, 0.5);
  faceAnchorGroup.add(box);
})

faceTracker.onVisible.bind(anchor => {
  faceAnchorGroupsArr[anchor.id].visible = true
})

faceTracker.onNotVisible.bind(anchor => {
  faceAnchorGroupsArr[anchor.id].visible = false
})

// Set up our render loop
function render() {
  requestAnimationFrame(render);
  camera.updateFrame(renderer);
  renderer.render(scene, camera);
}

requestAnimationFrame(render);
```

**Expected behavior**
All faces been tracked whenever the enter / leave screen.

**Desktop (please complete the following information):**
 - OS: MacOS Catalina 10.15.7
 - Browser Chrome
 - Version 92.0.4515.131

**Comment by nyan-left** [comment]
Thanks for the report, taking a look.

**Comment by nyan-left** [comment]
Patched the issue within `@zappar/zappar@0.3.12`. This fix should make it's way into @zappar/zappar-threejs shortly :)

**Comment by eric-beaufol** [comment]
Ok great, thanks @nyan-left  !

**Comment by nyan-left** [comment]
Fixed in [0.3.29](https://github.com/zappar-xr/zappar-threejs/releases/tag/0.3.29) 😃
